### PR TITLE
Add openai-php auto-instrumentation

### PR DESCRIPTION
**Pull request by cedricziel**
My goal here is to understand the interactions with OpenAI in more detail. I opted for internal spans, since the actual communication will be handled through an http client, but feel like this could be added on client spans as well, rewriting their names.

### CODE REVIEW

File: src/Instrumentation/OpenAIPHP/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[![Releases](https://img.shields.io/badge/releases-purple)](https://github.com/opentelemetry-php/contrib-auto-ext-amqp/releases)
+[![Issues](https://img.shields.io/badge/issues-pink)](https://github.com/open-telemetry/opentelemetry-php/issues)
+[![Source](https://img.shields.io/badge/source-contrib-green)](https://github.com/open-telemetry/opentelemetry-php-contrib/tree/main/src/Instrumentation/ExtAmqp)
+[![Mirror](https://img.shields.io/badge/mirror-opentelemetry--php--contrib-blue)](https://github.com/opentelemetry-php/contrib-auto-ext-amqp)
```
Comment (brettmc) [inline src/Instrumentation/OpenAIPHP/README.md:4]: Shall we name the gitsplit repo `contrib-auto-openai` ?

File: src/Instrumentation/OpenAIPHP/composer.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  "name": "open-telemetry/opentelemetry-auto-openai-php",
+  "description": "OpenTelemetry auto-instrumentation for openai-php/client",
+  "keywords": ["opentelemetry", "otel", "open-telemetry", "tracing", "io", "instrumentation", "openai"],
+  "type": "library",
+  "homepage": "https://opentelemetry.io/docs/php",
+  "readme": "./README.md",
+  "license": "Apache-2.0",
+  "minimum-stability": "dev",
+  "require": {
+    "php": "^8.2",
```
Comment (brettmc) [inline src/Instrumentation/OpenAIPHP/composer.json:11]: why 8.2? the client claims to work with 8.1

File: src/Instrumentation/OpenAIPHP/tests/Integration/OpenAIPHPInstrumentationTest.php [hunk 3]
```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Instrumentation\ExtAmqp\tests\Integration;
```
Comment (brettmc) [inline src/Instrumentation/OpenAIPHP/tests/Integration/OpenAIPHPInstrumentationTest.php:5]:
```suggestion
namespace OpenTelemetry\Tests\Instrumentation\OpenAIPHP\tests\Integration;
```

File: src/Instrumentation/OpenAIPHP/README.md (hunk 1 above)
Comment (cedricziel) [reply src/Instrumentation/OpenAIPHP/README.md:4]: sounds good!

File: src/Instrumentation/OpenAIPHP/composer.json (hunk 2 above)
Comment (cedricziel) [reply src/Instrumentation/OpenAIPHP/composer.json]: Lowered the requirement :)

File: src/Instrumentation/OpenAIPHP/_register.php [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+use OpenTelemetry\Contrib\Instrumentation\OpenAIPHP\OpenAIPHPInstrumentation;
+use OpenTelemetry\SDK\Sdk;
+
+if (class_exists(Sdk::class) && Sdk::isInstrumentationDisabled(OpenAIPHPInstrumentation::NAME) === true) {
+    return;
+}
+
+if (extension_loaded('opentelemetry') === false) {
+    trigger_error('The opentelemetry extension must be loaded in order to autoload the OpenTelemetry ext-amqp auto-instrumentation', E_USER_WARNING);
```
Comment (agoallikmaa) [inline src/Instrumentation/OpenAIPHP/_register.php:13]:
```suggestion
    trigger_error('The opentelemetry extension must be loaded in order to autoload the OpenTelemetry OpenAI auto-instrumentation', E_USER_WARNING);
```

File: src/Instrumentation/OpenAIPHP/_register.php [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    return;
+}
+
+if (!extension_loaded('amqp')) {
+    return;
+}
+
```
Comment (agoallikmaa) [inline src/Instrumentation/OpenAIPHP/_register.php:21]: Should not be necessary.
```suggestion
```

**Review by agoallikmaa — APPROVED** [review]

**Review by brettmc — APPROVED** [review]

**Comment by brettmc** [comment]
https://packagist.org/packages/open-telemetry/opentelemetry-auto-openai-php

**Pull request merged**